### PR TITLE
docs: update configuration examples on individual rule pages to use flat config

### LINKED ISSUE #9481: Docs: update configuration examples on individual rule pages to use flat config
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

The docs pages for individual rules — both on the current website, as well as the v8 website, show a config file snippet showing how to turn on the rule.

However, they reference `eslintrc.cjs`. These probably need to be updated with flat config.

Should we add the `Flat Config`/`Legacy Config` tabs [that we have on other pages](https://typescript-eslint.io/users/configs#projects-without-type-checking)?

### Affected URL(s)

https://typescript-eslint.io/rules/*****/
Example: https://typescript-eslint.io/rules/no-duplicate-type-constituents/

### PR DESCRIPTION
**Pull request by y-hsgw**
## PR Checklist

- [x] Addresses an existing open issue: fixes #9481 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

Added the `Flat Config`/`Legacy Config` tabs to the configuration examples on the rule pages.

📕

### CODE REVIEW

File: packages/website/plugins/generated-rule-docs/insertions/insertBaseRuleReferences.ts [hunk 1]
```diff
@@ -27,15 +30,57 @@ export function insertBaseRuleReferences(page: RuleDocsPage): string {
...
+          children: [
+            {
+              type: 'code',
+              lang: 'js',
+              meta: 'title="eslint.config.mjs"',
+              value: `export default tseslint.config(${getEslintConfigString(
+                extendsBaseRuleName,
+                page.file.stem,
+                true,
+              )})`,
```
Comment (JoshuaKGoldberg) [inline packages/website/plugins/generated-rule-docs/insertions/insertBaseRuleReferences.ts:55]: Missing a `;` to be consistent with the legacy configs & rest of the site:

```suggestion
              )});`,
```

File: packages/website/plugins/generated-rule-docs/insertions/insertNewRuleReferences.ts [hunk 2]
```diff
@@ -54,18 +54,65 @@ export async function insertNewRuleReferences(
...
+  const rules = `{
     "@typescript-eslint/${page.file.stem}": "error"
-  }
+  }`;
+
+  const eslintrc = `{
+  "rules": ${rules}
+}`;
+
+  const eslitnConfig = `{
```
Comment (JoshuaKGoldberg) [inline packages/website/plugins/generated-rule-docs/insertions/insertNewRuleReferences.ts:65]:
```suggestion
  const eslintConfig = `{
```

File: packages/website/plugins/utils/rules.ts [hunk 3]
```diff
@@ -17,24 +17,45 @@ export const eslintPluginDirectory = path.resolve(
...
+/**
+ * @param withComment Whether to include a full comment note.
+ */
+export function getEslintConfigString(
+  extendsBaseRuleName: string,
+  stem: string,
+  withComment: boolean,
+): string {
+  return `{
+  rules: ${getRulesString(extendsBaseRuleName, stem, withComment)}
```
Comment (JoshuaKGoldberg) [inline packages/website/plugins/utils/rules.ts:58]: [Style] Both `getEslintrcString` and `getEslintConfigString` do very little other than wrap `getRulesString`. Is there any reason they can't be inlined?

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
LGTM! Just a few small things. Thanks!

File: packages/website/plugins/generated-rule-docs/insertions/insertBaseRuleReferences.ts (hunk 1 above)
Comment (y-hsgw) [reply packages/website/plugins/generated-rule-docs/insertions/insertBaseRuleReferences.ts]: I fixed it in [c2aad1c](https://github.com/typescript-eslint/typescript-eslint/pull/10164/commits/c2aad1c8b0c321cb51ae4a31dd0d91b96a0a8beb).

File: packages/website/plugins/generated-rule-docs/insertions/insertNewRuleReferences.ts (hunk 2 above)
Comment (y-hsgw) [reply packages/website/plugins/generated-rule-docs/insertions/insertNewRuleReferences.ts]: I fixed it in [c2aad1c](https://github.com/typescript-eslint/typescript-eslint/pull/10164/commits/c2aad1c8b0c321cb51ae4a31dd0d91b96a0a8beb).

File: packages/website/plugins/utils/rules.ts (hunk 3 above)
Comment (y-hsgw) [reply packages/website/plugins/utils/rules.ts]: You're absolutely right! I was too focused on consolidating it.
I fixed it in [c2aad1c](https://github.com/typescript-eslint/typescript-eslint/pull/10164/commits/c2aad1c8b0c321cb51ae4a31dd0d91b96a0a8beb).

**Review by JoshuaKGoldberg — APPROVED** [review]
🙌 yay, thanks!

**Pull request merged**